src/claylib.lisp

```diff
@@ -39,19 +39,37 @@ background of the render texture, or NIL to skip clearing."
      ,@body
      (end-texture-mode)))
 
+;; FIXME Problem with this approach: entering a nested do-game-loop will overwrite *SCENE*
```

Comment:
This can be fixed by declaring `*scene*` as special. (May need a separate declaration for every loop invocation.)

Reply:
Ah right... I had forgotten that's what special does. Never actually used it since learning about it.

Reply:
Update: defvar/defparameter already declare stuff as special. The trick is not to  `setf *scene*` without first shadowing it. So we can just use `let` in `do-game-loop` and then any `setf`s in that scope (such as the one in `switch-scene`) will play nicely with dynamic binding.